subsys/bluetooth/host/Kconfig.l2cap

```diff
@@ -50,6 +52,7 @@ config BT_L2CAP_TX_MTU
 
 config BT_L2CAP_DYNAMIC_CHANNEL
 	bool "L2CAP Dynamic Channel support"
+	default y
```

Comment:
How come?

Reply:
I was already surprised that EATT defaulted to `y` but thought it's not that bad since it depends on DYNAMIC_CHANNEL, but I now see you default that also to `y` which pulls everything in unless explicitly disabled, which doesn't seem wise.

Reply:
Right, we can have EATT select BT_L2CAP_DYNAMIC_CHANNEL instead of defaulting to y.

Reply:
Looks like we cannot really use select:

```
warning: BT_L2CAP_DYNAMIC_CHANNEL (defined at subsys/bluetooth/host/Kconfig.l2cap:53) has direct dependencies BT_SMP && BT_CONN && BT_HCI_HOST && BT_HCI && BT with value n, but is currently being y-selected by the following symbols:
 - BT_EATT (defined at subsys/bluetooth/host/Kconfig.gatt:37), with value y, direct dependencies BT_CONN && BT_HCI_HOST && BT_HCI && BT (value: y), and select condition BT_CONN && BT_HCI_HOST && BT_HCI && BT (value: y)
```

The other option is left EATT as optional, though I guess later on we would like to have EATT enabled by default if BT_L2CAP_DYNAMIC_CHANNEL is enabled, so perhaps we can use default y if BT_L2CAP_DYNAMIC_CHANNEL.

Reply:
So I resolve this by having a dependency to BT_L2CAP_DYNAMIC_CHANNEL in addition to default y if BT_L2CAP_DYNAMIC_CHANNEL.